File: Python/Easy/278. First Bad Version.py
# https://leetcode.com/problems/first-bad-version/description/

# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        left=1
        right=n
        while(left<right):
            m=(left+right)//2
            if(isBadVersion(m)):
                right=m
            else:
                left=m+1
        return left
        

File: C++/Easy/278. First Bad Version.cpp
// https://leetcode.com/problems/first-bad-version/description/

// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int left = 1;
        int right = n;
        while (left < right) {
            int m = left + (right - left) / 2;
            if (isBadVersion(m)) {
                right = m;
            } else {
                left = m + 1;
            }
        }
        return left;
    }
};